### PR TITLE
Don't pick targets which aren't built yet. (#663).

### PR DESCRIPTION
**Pull request by oliverchang**

### CODE REVIEW

File: src/python/google_cloud_utils/storage.py [hunk 1]
```diff
@@ -325,19 +337,23 @@ def _get_object_properties(self, remote_path):
...
-        'size':
-            os.path.getsize(fs_path),
-        'metadata':
-            self._get_metadata(bucket, path),
+    data = {
+        'bucket': bucket,
+        'name': path,
     }
 
+    if not os.path.isdir(fs_path):
```
Comment (inferno-chromium) [inline src/python/google_cloud_utils/storage.py]: Can you add a comment on why we don't send these for directories. Is it ok to generalize ?

File: src/python/google_cloud_utils/storage.py [hunk 2]
```diff
@@ -188,14 +188,26 @@ def list_blobs(self, remote_path):
...
+    iterator = bucket.list_blobs(prefix=path, delimiter=delimiter)
+    for blob in iterator:
       properties['bucket'] = bucket_name
       properties['name'] = blob.name
       properties['updated'] = blob.updated
       properties['size'] = blob.size
 
       yield properties
 
+    if not recursive:
```
Comment (inferno-chromium) [inline src/python/google_cloud_utils/storage.py]: Can you explain this block more in a comment, is it for dirs ?

File: src/python/build_management/build_manager.py [hunk 3]
```diff
@@ -1154,13 +1154,18 @@ def _get_targets_list(bucket_path):
...
+  targets_list_path = os.path.join(bucket_dir_path, TARGETS_LIST_FILENAME)
   data = storage.read_data(targets_list_path)
   if not data:
     return None
 
-  return data.splitlines()
+  # Filter out targets which are not yet built.
+  targets = data.splitlines()
+  listed_targets = set(
+      os.path.basename(path.rstrip('/'))
```
Comment (inferno-chromium) [inline src/python/build_management/build_manager.py]: shouldn't list_blobs strip the / or is too problematic there ?

File: src/python/google_cloud_utils/storage.py (hunk 1 above)
Comment (oliverchang) [reply src/python/google_cloud_utils/storage.py]: added a comment.

File: src/python/google_cloud_utils/storage.py (hunk 2 above)
Comment (oliverchang) [reply src/python/google_cloud_utils/storage.py]: added a comment.

File: src/python/build_management/build_manager.py (hunk 3 above)
Comment (oliverchang) [reply src/python/build_management/build_manager.py]: I think it's good to leave the '/'. It's what `gsutil ls` returns.

File: src/python/google_cloud_utils/storage.py [hunk 4]
```diff
@@ -188,14 +188,27 @@ def list_blobs(self, remote_path):
...
     properties = {}
 
-    for blob in bucket.list_blobs(prefix=path):
+    if recursive:
+      delimiter = None
+    else:
+      delimiter = '/'
+
+    iterator = bucket.list_blobs(prefix=path, delimiter=delimiter)
+    for blob in iterator:
```
Comment (Dor1s) [inline src/python/google_cloud_utils/storage.py]: Is it safe to rely on this behavior regarding delimiter presence? Maybe adding an extra check for `if recursive` wouldn't hurt?

**Pull request merged**